doc/api/url.md

```diff
@@ -921,6 +921,59 @@ console.log(url.format(myURL, { fragment: false, unicode: true, auth: false }));
 // Prints 'https://你好你好/?abc'
 ```
 
+### url.pathToFileURL(path)
+
+* `path` {string} The absolute path to convert to a File URL.
+* Returns: {URL} The file URL object.
+
+This function ensures the correct encodings of URL control characters in file
+paths when converting into File URLs.
+
+For example, the following errors can occur when converting from paths to URLs:
+
+```js
+// throws for missing schema (posix)
+// (in Windows the drive letter is detected as the protocol)
+new URL(__filename);
+
+// 'file:///foo' instead of the correct 'file:///foo%231' (posix)
+new URL('./foo#1', 'file:///');
+
+// 'file:///nas/foo.txt' instead of the correct 'file:///foo.txt' (posix)
+new URL(`file://${'//nas/foo.txt'}`);
+
+// 'file:///some/path%' instead of the correct 'file:///some/path%25' (posix)
+new URL(`sfile:${'/some/path%.js'}`);
+```
+
+where using `pathToFileURL` we can get the correct results above.
+
+### url.fileURLToPath(url)
```

Comment:
This section needs to go before `### url.format(URL[, options])` ABC-wise.

Reply:
Sure, a shame we can't keep these together though.